python/common/org/python/types/Float.java

```diff
@@ -169,7 +171,15 @@ public Float(double value) {
         __doc__ = ""
     )
     public org.python.Object __ne__(org.python.Object other) {
-        throw new org.python.exceptions.NotImplementedError("float.__ne__() has not been implemented.");
+        if (other instanceof org.python.types.Int || other instanceof org.python.types.Float ||
```

Comment:
Hey, yes. Float and string datatypes are comparable. However, it would always result in False. I'm on Python 3.4 and it gives the same results as yours.
Also, what do you mean that the string test was passing? As far as I can see, the test_ne_str was in the not_implemented list.
I've added code to include the string datatype in my implementation now.
Thank you for pointing it out!